tests/unit_tests/array/CMakeLists.txt

```diff
@@ -1,22 +1,30 @@
-enable_testing()
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
 
-foreach(etest ${all_unit_tests})
-    add_detray_test(array_${etest}
-                    array_${etest}.cpp
-                    LINK_LIBRARIES
-                    detray::array)
-endforeach(etest)
+# Set up all of the "standard" tests.
+detray_add_test( array
+   "array_annulus2.cpp" "array_core.cpp" "array_cylinder_intersection.cpp"
+   "array_cylinder3.cpp" "array_detector.cpp" "array_geometry_graph.cpp"
+   "array_geometry_linking.cpp" "array_geometry_navigation.cpp"
+   "array_geometry_scan.cpp" "array_intersection_kernel.cpp"
```

Comment:
They are built into a single executable. (`detray_test_array`) The source files are still compiled into separate object files. Which are linked into a single executable in the end. So... regular build system stuff.

Some of these tests are indeed slower than the others, especially in Debug mode. But I myself usually want to run as many unit tests at the same time as possible. Since I want to test that everything is still working after my changes. Having to use `ctest` to run all of the tests is doable, but it's yet another indirection while debugging the code.

So all in all, I'm quite fond of building large test executables. You can still filter what you want to run using GoogleTest's command line arguments. But if you absolutely insist on separating this into more executables, that can of course be done.